### PR TITLE
robustify `TkHistoMap` and related unit tests

### LINKED ISSUE #39455: Insufficient error handling in TkHistoMap::fillFromAscii
In various pull request tests we are seeing segmentation faults originating from `TkHistoMap::fillFromAscii`. Looking at the implementation, there is an error case which could be causing this if there is a problem reading the file. See below for details.

### PR DESCRIPTION
**Pull request by mmusich**
resolves https://github.com/cms-sw/cmssw/issues/39455

#### PR description:

Title says it all, see analysis at https://github.com/cms-sw/cmssw/issues/39455#issuecomment-1252551145.
Supply also a real input file for the unit test.

#### PR validation:

Relies on the unit tests.
Run successfully:

```console
scram b runtests_test_TkHistoMap 
```

#### If this PR is a backport please specify the original PR and why you need to backport that PR. If this PR will be backported please specify to which release cycle the backport is meant for:

N/A

### CODE REVIEW

**Comment by mmusich** [comment]
type trk, bug-fix

File: DQM/SiStripCommon/test/plugins/testTkHistoMap.cc [hunk 1]
```diff
@@ -48,13 +50,18 @@ class testTkHistoMap : public DQMOneEDAnalyzer<> {
...
   std::unique_ptr<TkHistoMap> tkhistoID, tkhisto, tkhistoBis, tkhistoZ, tkhistoPhi, tkhistoR, tkhistoCheck;
 };
 
 //
 testTkHistoMap::testTkHistoMap(const edm::ParameterSet& iConfig)
-    : tTopoToken(esConsumes()), tkGeoToken(esConsumes()), readFromFile(iConfig.getParameter<bool>("readFromFile")) {}
+    : tTopoToken(esConsumes()),
+      tkGeoToken(esConsumes()),
+      readFromFile_(iConfig.getParameter<bool>("readFromFile")),
+      fp_(iConfig.getUntrackedParameter<edm::FileInPath>("inputFile",
```
Comment (Dr15Jones) [inline DQM/SiStripCommon/test/plugins/testTkHistoMap.cc:63]: This should probably be a tracked parameter as the `readFromFile` is tracked. You can give defaults for tracked parameter if you use a `fillDescription` function.

**Comment by Dr15Jones** [comment]
@mmusich thanks for the quick fix!

**Comment by Dr15Jones** [comment]
please test

**Comment by tvami** [comment]
+alca
 * test pass, PR according to the desc

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1

**Pull request merged**